proc_CPMG.py

```diff
@@ -81,23 +73,10 @@ def print_fun(x, f, accepted):
     #{{{select echo decay fit function
     data = s['t2':(-200,200)].sum('t2')
     fl.next('Echo decay')
-    x = s.getaxis('nEchoes')
-    ydata = data.data.real
-    ydata /= max(ydata)
-    fl.plot(x,ydata,'-o', alpha=0.7, label='%s'%label_str, human_units=False)
-    fitfunc = lambda p, x: p[0]*exp(-x*p[1])
-    errfunc = lambda p_arg, x_arg, y_arg: fitfunc(p_arg, x_arg) - y_arg
-    p0 = [0.1,100.0,-3.0]
-    p1, success = leastsq(errfunc, p0[:], args=(x, ydata))
-    assert success == 1, "Fit did not succeed"
+    raise RuntimeError("set up nonlinear fitting of T2 decay with "+str([0.1,100.0,-3.0])+"as a guess")
```

Comment:
I'm assuming this is a "to do" for me? and not using fitdata function? or this is where the fitdata function would go?

Reply:
yes, this is a to-do

Reply:
an we are using fitdata for this?

Reply:
this is done now

Reply:
I made some changes that will break this -- please inspect + re-fix w/ improvements

Reply:
(resolve when you read this)

Reply:
(what you had there was good, but there is a problem with the units for the echo number -- it needs to be a time, not a number -- this is how you were getting weird T2's at some point)